### PR TITLE
Array property type UX improvements

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR is a follow-up of #1513 and adds UX improvements to Array property types creation.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203312852763953/1203366629132086/f) _(internal)_
- [Asana task](https://app.asana.com/0/1203312852763953/1203366629132089/f) _(internal)_

## 🔍 What does this change?

- Added the turquoise color to theme and gray[100] color which was missing (colors taken from [here](https://www.figma.com/file/xtjsETSOXMf2bgtcBL13Bh/HASH-Global-Styles?node-id=1643%3A108&t=o4oHXkiV5Jo5M6Kb-4))
- Added cubes icon for property object array;
- Added an `arrayType` property to array expected values to track what kind of array the property is.
- Added a "close" icon on data type badges that remove them from the array (confirmation modal not implemented in this PR);
- Implemented language changes according to expected types ("Containing", "Containing an", etc..);
- Added an edit  button to the pills of array data types, clicking it opens the modal with the type information;
- Added a close buttons to data type pills;

## 📹 Demo

image

image

### CODE REVIEW

**Review by CiaranMn — COMMENTED** [review]
I have tested this – I haven't looked at the diff in detail, but I've added a few comments.

Can we now add the badge for arrays to the expected values column in the property type table? It's currently empty for array-only expected types

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/array-property-type-menu.tsx [hunk 1]
```diff
@@ -59,28 +63,41 @@ export const ArrayPropertyTypeMenu: FunctionComponent<
     name: `flattenedPropertyList.${id}.data.expectedValues`,
   });
 
+  const deleteDataType = (typeId: string) => {
+    const removedPropertyId = Object.values(flattenedProperties).find(
```
[the remark this answers is not in the document]
Comment (luisbettencourt) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/array-property-type-menu.tsx]: Renamed in 21114859984a38acb718d22f0f7009bddee53c4d.

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/array-property-type-menu.tsx [hunk 2]
```diff
@@ -59,28 +63,41 @@ export const ArrayPropertyTypeMenu: FunctionComponent<
     name: `flattenedPropertyList.${id}.data.expectedValues`,
   });
 
+  const deleteDataType = (typeId: string) => {
```
[the remark this answers is not in the document]
Comment (luisbettencourt) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/array-property-type-menu.tsx]: Renamed to `deleteDataTypeByTypeId´ in 4efdf14001b5d57dc046efe7548dddfbcbf95a1e.

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/array-property-type-menu.tsx [hunk 3]
```diff
@@ -41,12 +42,15 @@ const deletePropertyAndChildren = (
 
 type ArrayPropertyTypeMenuProps = {
   id: string;
```
[the remark this answers is not in the document]
Comment (luisbettencourt) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/array-property-type-menu.tsx]: After some reflection I realized most variables named with a variation of "property" were incorrect and should be "dataType" instead. Addressed this and a lot of other renaming changes in a073c82db98419f9016a955a1fd5d37c213289ab.

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/array-property-type-menu.tsx [hunk 4]
```diff
@@ -100,41 +117,53 @@ export const ArrayPropertyTypeMenu: FunctionComponent<
...
+          const isObject =
+            property.data.typeId === types.dataType.object.dataTypeId;
+
+          return (
+            <Box key={property.data.typeId} mb={1}>
+              {property.data.typeId === "array" ? (
+                <ArrayPropertyTypeMenu
+                  id={childId}
+                  prefix={
+                    expectedValues.length === 1 || pos === 0
```
[the remark this answers is not in the document]
Comment (luisbettencourt) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/array-property-type-menu.tsx]: Good catch! 7ffdbb9d0146753f87fd1d482ef7ef378d27e650

**Pull request merged**